pvlib/test/test_iam.py

```diff
@@ -99,11 +100,59 @@ def test_martin_ruiz():
     iam = _iam.martin_ruiz(aoi, a_r)
     assert_series_equal(iam, expected)
 
-    # check exception clause
-    with pytest.raises(RuntimeError):
+
+def test_martin_ruiz_exception():
+
+    with pytest.raises(ValueError):
         _iam.martin_ruiz(0.0, a_r=0.0)
 
 
+def test_martin_ruiz_diffuse():
+
+    surface_tilt = 30.
+    a_r = 0.16
+    expected = (0.9549735, 0.7944426)
+
+    # will fail if default values change
+    iam = _iam.martin_ruiz_diffuse(surface_tilt)
+    assert_allclose(iam, expected)
+
+    # will fail if parameter names change
+    iam = _iam.martin_ruiz_diffuse(surface_tilt=surface_tilt, a_r=a_r)
+    assert_allclose(iam, expected)
+
+    a_r = 0.18
+    surface_tilt = [0, 30, 90, 120, -30, np.nan, np.inf]
+    expected_sky = [0.9407678, 0.9452250, 0.9407678, 0.9055541, 0.9452250,
```

Comment:
`surface_tilt` = -30 should not produce a value that appears valid.

Reply:
Is there consensus on this?